File: Examples/CSharp/GroupDocs.Conversion.Examples.CSharp/CustomInputDataHandler.cs
using System.IO;
using Amazon;
using Amazon.S3;
using Amazon.S3.Model;
using GroupDocs.Conversion.Domain;
using GroupDocs.Conversion.Handler.Input;

namespace GroupDocs.Conversion.Examples.CSharp
{
    class CustomInputDataHandler : IInputDataHandler
    {
        private static string bucketName = ""; //TODO: Put you bucketname here 
        private readonly AmazonS3Client _client;
        public CustomInputDataHandler()
        {
            _client = new AmazonS3Client(RegionEndpoint.EUWest1);
        }
        public FileDescription GetFileDescription(string guid)
        {
            FileDescription result = new FileDescription();
            var request = new GetObjectRequest
            {
                BucketName = bucketName,
                Key = guid
            };
            string fileName;
            long size;
            using (var response = _client.GetObject(request))
            {
                fileName = response.Key;
                size = response.ContentLength;
            }

            result.Guid = guid;
            result.Name = fileName;
            result.Size = size;
            return result;
        }
        public Stream GetFile(string guid)
        {
            var request = new GetObjectRequest
            {
                BucketName = bucketName,
                Key = guid
            };
            var result = new MemoryStream();
            using (var response = _client.GetObject(request))
            {
                byte[] buffer = new byte[16384]; //16*1024
                int read;
                while ((read = response.ResponseStream.Read(buffer, 0, buffer.Length)) > 0)
                {
                    result.Write(buffer, 0, read);
                }
            }
            return result;
        }
    }
}

File: Examples/VisualBasic/GroupDocs.Conversion.Examples.VisualBasic/CustomInputDataHandler.vb
'ExStart:CustomInputDataHandlerClass
Imports System.IO
Imports Amazon
Imports Amazon.S3
Imports Amazon.S3.Model
Imports GroupDocs.Conversion.Domain
Imports GroupDocs.Conversion.Handler.Input

Namespace GroupDocsConversionExamples.VisualBasic
    Class CustomInputDataHandler
        Implements IInputDataHandler
        Private Shared bucketName As String = ""
        'TODO: Put you bucketname here 
        Private ReadOnly _client As AmazonS3Client
        Public Sub New()
            _client = New AmazonS3Client(RegionEndpoint.EUWest1)
        End Sub
        Public Function GetFileDescription(guid As String) As FileDescription Implements IInputDataHandler.GetFileDescription
            Dim result As New FileDescription()
            Dim requests = New GetObjectRequest()
            requests.BucketName = bucketName
            requests.Key = guid
            Dim fileName As String
            Dim size As Long
            Using response = _client.GetObject(requests)
                fileName = response.Key
                size = response.ContentLength
            End Using

            result.Guid = guid
            result.Name = fileName
            result.Size = size
            Return result
        End Function
        Public Function GetFile(guid As String) As Stream Implements IInputDataHandler.GetFile
            Dim requests As New GetObjectRequest()
            requests.BucketName = bucketName
            requests.Key = guid

            Dim result = New MemoryStream()
            Using response = _client.GetObject(requests)
                Dim buffer As Byte() = New Byte(16383) {}
                '16*1024
                Dim read As Integer
                While (InlineAssignHelper(read, response.ResponseStream.Read(buffer, 0, buffer.Length))) > 0
                    result.Write(buffer, 0, read)
                End While
            End Using
            Return result
        End Function
        Private Shared Function InlineAssignHelper(Of T)(ByRef target As T, value As T) As T
            target = value
            Return value
        End Function
    End Class
End Namespace
'ExEnd:CustomInputDataHandlerClass
